### DISCUSSION TITLE
Popup tooltip reference on mouse over

### DISCUSSION THREAD

**Discussion by FabriceNeyret** [DesModder/DesModder#840] [Ideas]
Desmos should have do that, but...

Especially for beginners, but also because some functions have many variants and options, it could be useful when typing / mouseovering a keyword to have a popup tooltip telling the possible parameters (and possibly a short definition too, maybe).

**Comment by jared-hughes** [comment]
I love this idea! Especially for `sort`, `arctan`, and other functions which people don't realize have a two argument form.

Maybe this could also be extended to a bit of intellisense, where hovering over a variable could show the current value of the variable. Or more generally, highlighting a portion of the formula could evaluate it.

**Comment by FabriceNeyret** [comment]
plenty of commands have optional param 1,2,3 (e.g. random) or can apply to [lists] or v1,v2,v3... and not only 1 value (e.g. max).

yeah, I didn't dare suggesting more info (or to display it that way), but sure there are more things usefull to display on demand: value of variables (especially for lists where it's uneasy to see), finding where or how a variable is defined or its range. From the expression part, put potentially from a graph element too (at least the name, or highlighting the equation or folder in the code ) .

**Comment by jared-hughes** [comment]
When you click a displayed curve, it selects the corresponding equation already.

Or do you mean showing the corresponding equation on hover as something like a tooltip? I think that would be cool. Maybe right-click a curve to directly show the latex of the corresponding equation.

**Comment by FabriceNeyret** [comment]
"shows already": not always. not for implicits, for instance.

**Comment by jared-hughes** [comment]
For implicits, you have to be careful with clicking specifically the edge of the curve; clicking the interior does not work (probably because that would cause issues with dragging the graphpaper when a region covers the whole graphpaper) (see https://www.desmos.com/calculator/hizni99e50; this still applies when the region's lines are disabled).

**Comment by FabriceNeyret** [comment]
sorry, I meant parametrics , like (cos(t),sin(t))

**Comment by jared-hughes** [comment]
Oh interesting, that seems like a bug with Desmos

**Comment by jared-hughes** [comment]
In addition to showing a description of builtin functions on hover, it would be good to allow users to specify a type signature and a python docstring-style description of each function. (Discussed slightly in #83)
